elftools/dwarf/dwarfinfo.py

```diff
@@ -164,31 +164,42 @@ def has_CFI(self):
         """
         return self.debug_frame_sec is not None
 
-    def CFI_entries(self):
-        """ Get a list of dwarf_frame CFI entries from the .debug_frame section.
+    @lazy
+    def cfi(self):
+        """ Wrapper for .debug_frame section.
         """
-        cfi = CallFrameInfo(
+        return CallFrameInfo(
             stream=self.debug_frame_sec.stream,
             size=self.debug_frame_sec.size,
             address=self.debug_frame_sec.address,
             base_structs=self.structs)
-        return cfi.get_entries()
+
+    def CFI_entries(self):
```

Comment:
It's not clear to me what is gained from this new laziness of CFI entries, for example. If a callsite calls `CFI_entries`, the parsing happens only on the first call, and isn't repeated? But why would `CFI_entries` be called multiple times in the first place?

Reply:
That patch only prevents creation of multiple `CallFrameInfo` instances.
Entries parsing is already lazy and its result cached internally in `CallFrameInfo`.
Because previous `*CFI_entries` method created new `CallFrameInfo` instance each time, the caching has not been used actually.